src/OpenApi/gen/XmlCommentGenerator.Emitter.cs

```diff
@@ -0,0 +1,392 @@
+                }
+                catch (JsonException)
+                {
+                    return null;
+                }
+            }
+        }
+    }
+
+    {{GeneratedCodeAttribute}}
+    file static class GeneratedServiceCollectionExtensions
+    {
+{{GenerateAddOpenApiInterceptions(groupedAddOpenApiInvocations)}}
+    }
+}
+""";
+
+    internal static string GetAddOpenApiInterceptor(AddOpenApiOverloadVariant overloadVariant) => overloadVariant switch
+    {
+        AddOpenApiOverloadVariant.AddOpenApi => """
+        public static IServiceCollection AddOpenApi(this IServiceCollection services)
+                {
+                    return services.AddOpenApi("v1", options =>
+                    {
+                        options.AddSchemaTransformer(new XmlCommentSchemaTransformer());
+                        options.AddOperationTransformer(new XmlCommentOperationTransformer());
+                    });
+                }
+        """,
+        AddOpenApiOverloadVariant.AddOpenApiDocumentName => """
+        public static IServiceCollection AddOpenApi(this IServiceCollection services, string documentName)
+                {
+                    return services.AddOpenApi(documentName, options =>
+                    {
+                        options.AddSchemaTransformer(new XmlCommentSchemaTransformer());
+                        options.AddOperationTransformer(new XmlCommentOperationTransformer());
+                    });
+                }
+        """,
+        AddOpenApiOverloadVariant.AddOpenApiConfigureOptions => """
+        public static IServiceCollection AddOpenApi(this IServiceCollection services, Action<OpenApiOptions> configureOptions)
+                {
+                    return services.AddOpenApi("v1", options =>
+                    {
+                        configureOptions(options);
+                        options.AddSchemaTransformer(new XmlCommentSchemaTransformer());
+                        options.AddOperationTransformer(new XmlCommentOperationTransformer());
+                    });
+                }
+        """,
+        AddOpenApiOverloadVariant.AddOpenApiDocumentNameConfigureOptions => """
+        public static IServiceCollection AddOpenApi(this IServiceCollection services, string documentName, Action<OpenApiOptions> configureOptions)
+                {
+                    // This overload is not intercepted.
+                    return OpenApiServiceCollectionExtensions.AddOpenApi(services, documentName, options =>
+                    {
+                        configureOptions(options);
+                        options.AddSchemaTransformer(new XmlCommentSchemaTransformer());
+                        options.AddOperationTransformer(new XmlCommentOperationTransformer());
+                    });
+                }
+        """,
+        _ => throw new InvalidOperationException("Invalid overload variant for `AddOpenApi`.")
```

Comment:
Maybe include the signature of the method including the parameters (e.g. `AddOpenApi(Foo bar)`)?